db-engine/src/main/java/com/splicemachine/db/iapi/types/SqlXmlUtil.java

```diff
@@ -670,12 +670,11 @@ protected List evalXQExpression(XMLDataValue xmlContext,
                 }
                 break;
             default:
+                String msg = "Don't know how to handle XPath result type " + result.getResultType();
                 if (SanityManager.DEBUG) {
-                    SanityManager.THROWASSERT(
-                            "Don't know how to handle XPath result type " +
-                            result.getResultType());
+                    SanityManager.THROWASSERT(msg);
                 }
-                itemRefs = null;
+                throw new Exception(msg);
```

Comment:
Could you raise an IllegalArgumentException or UnsupportedOperationException, rather than a generic Exception ?

Reply:
Right, throw an UnsupportedOperationException is better.